slsDetectorSoftware/src/slsDetector.cpp

```diff
@@ -3888,6 +3865,32 @@ void slsDetector::setPipeline(int clkIndex, int value) {
     sendToDetector(F_SET_PIPELINE, args, nullptr);
 }
 
+void slsDetector::setCounters(uint32_t countermask) {
+    FILE_LOG(logDEBUG1) << "Setting Counters to " << countermask;
+    sendToDetector(F_SET_COUNTERS, countermask, nullptr);
+    sendCounterstoReceiver(countermask);
+}
+
+void slsDetector::sendCounterstoReceiver(uint32_t countermask) {
+    if (shm()->useReceiverFlag) {
```

Comment:
Speedwise it's not needed but just for fun, many CPUs come with a special instruction to count bits in an integer. This is accessible through int __builtin_popcount (unsigned int x) on both clang and gcc. 

Just for fun you can compare the length of the assembly code. 
[https://godbolt.org/z/QuY4Xu](https://godbolt.org/z/QuY4Xu)

Here there is no speed advantage but replacing the loop with the popcount function means less code to worry about.

Reply:
Sounds great! Changed it to the popcounter :D